master/docs/manual/configuration/changesources.rst

```diff
@@ -1002,6 +1002,11 @@ The :bb:chsrc:`HgPoller` accepts the following arguments:
     Set encoding will be used to parse author's name and commit message.
     Default encoding is ``'utf-8'``.
 
+``revlink``
+    A function that maps branch and revision to a valid url (e.g. hgweb), stored along with the change.
+    This function must be a callable which takes two arguments, the branch and the revision.
+    Defaults to lambda branch, revision: (u'')
```

Comment:
Copied from p4 actually since it behaves the same.

I suppose it could default to `repourl + "/rev/{}"` if one assumes that `repourl` is a hgweb instance. Not sure if it would be a good idea to make it conditional on `repourl` having a `/^https?/` prefix, I've heard people get confused by conditional behavior like that in the past. Thoughts?

Reply:
I think conditional is fine as long as it is documented.
Actually, after more read, I understand this default.  It wasn't clear to me that `defaults is always returning empty string`